lib/CustomSurvey.dart

```diff
@@ -0,0 +1,235 @@
+        actions: [
+          IconButton(
+            onPressed: context.watch<MobileEditing>().state ? null : validate,
+            icon: const Icon(Icons.save),
+          ),
+        ],
+      ),
+      body: Center(
+        child: SingleChildScrollView(
+          child: Column(
+            children: [
+              ReorderableListView(
+                physics: const NeverScrollableScrollPhysics(),
+                shrinkWrap: true,
+                buildDefaultDragHandles: false,
+                onReorder: (oldIndex, newIndex) {
+                  if (newIndex > oldIndex) newIndex--;
+                  setState(() {
+                    keyedQuestions.insert(newIndex, keyedQuestions.removeAt(oldIndex));
+                  });
+                },
+                children: editors,
+              ),
+              divider(keyedQuestions.length),
+            ],
+          ),
+        ),
+      ),
+      floatingActionButton: editButton,
+    );
+  }
+}
+
+class SurveyEditDivider extends StatefulWidget {
+  const SurveyEditDivider(this.addQuestion, {Key? key}) : super(key: key);
+  final void Function(SurveyQuestion) addQuestion;
+
+  static const height = 60.0;
+
+  @override
+  State<SurveyEditDivider> createState() => _SurveyEditDividerState();
+}
+
+class _SurveyEditDividerState extends State<SurveyEditDivider> {
+  final node = FocusNode();
+  bool hovered = false, focused = false;
+
+  @override
+  void initState() {
+    super.initState();
+    node.addListener(() => setState(() => focused = node.hasPrimaryFocus));
+  }
+
+  @override
+  void dispose() {
+    node.dispose();
+    super.dispose();
+  }
+
+  static const presets = [
+    YesNoQuestion('[question]'),
+    TextPromptQuestion('[question]'),
+    RadioQuestion('[question]'),
+    CheckboxQuestion('[question]'),
+    ScaleQuestion('[question]'),
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    if (context.watch<MobileEditing>().state) {
+      return const SizedBox(height: SurveyEditDivider.height / 2);
+    }
+    final Widget button;
+    if (focused) {
+      button = Focus(
+        focusNode: node,
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            for (final question in presets) ...[
+              InkWell(
+                onTap: () {
+                  widget.addQuestion(question);
+                  node.unfocus();
+                },
+                child: QuestionTypeIcon(question),
+              ),
+            ],
+          ],
+        ),
+      );
+    } else {
+      button = InkWell(
+        splashColor: Colors.transparent,
+        focusColor: Colors.transparent,
+        focusNode: node,
+        onTap: node.requestFocus,
+        child: Padding(
+          padding: EdgeInsets.all(hovered ? 8 : 2),
+          child: Opacity(
+            opacity: hovered ? 1 : 0.5,
+            child: const Icon(Icons.add),
+          ),
+        ),
+      );
+    }
+
+    final
```

Comment:
It looks like [the file you uploaded](https://github.com/theheartcenter-inc/thc/blob/7167e386483e63949a526e9b4bab043ff44c8aab/lib/CustomSurvey.dart#L235) was cut off about halfway through.

![half the file](https://github.com/theheartcenter-inc/thc/assets/10457200/e443fafb-8988-4594-a9da-945b17f7be94)

The good news is that I'm ready to start my [**Admin: view & edit surveys**](https://www.notion.so/Admin-view-edit-surveys-e4c2ea6e53e544efb0d126d49ebc9fa5) task, and the `sendCustomSurveyToFirestore()` gives me a great starting point!

Instead of merging into `main`, let's incorporate these changes into the "admin portal survey stuff" branch.